power_grid_model_c/power_grid_model/include/power_grid_model/component/asym_line.hpp

```diff
@@ -0,0 +1,115 @@
+
+        y_series = inv(z_series);
+        y_shunt = 2 * pi * system_frequency * c_matrix * 1.0i;
+    }
+
+    // override getter
+    constexpr double base_i_from() const override { return base_i_; }
+    constexpr double base_i_to() const override { return base_i_; }
+    constexpr double loading(double /* max_s */, double max_i) const override { return max_i / i_n_; };
+    constexpr double phase_shift() const override { return 0.0; }
+    constexpr bool is_param_mutable() const override { return false; }
+
+  private:
+    double i_n_;
+    double base_i_;
+    ComplexTensor<asymmetric_t> y_series;
+    ComplexTensor<asymmetric_t> y_shunt;
+
+    ComplexTensor<asymmetric_t> compute_z_series_from_input(const power_grid_model::AsymLineInput& asym_line_input) {
+        ComplexTensor<asymmetric_t> z_series_abc;
+        if (is_nan(asym_line_input.r_na) && is_nan(asym_line_input.x_na)) {
+            ComplexTensor<asymmetric_t> r_matrix = ComplexTensor<asymmetric_t>(asym_line_input.r_aa, asym_line_input.r_bb, asym_line_input.r_cc, asym_line_input.r_ba, asym_line_input.r_ca, asym_line_input.r_cb);
+            ComplexTensor<asymmetric_t> x_matrix = ComplexTensor<asymmetric_t>(asym_line_input.x_aa, asym_line_input.x_bb, asym_line_input.x_cc, asym_line_input.x_ba, asym_line_input.x_ca, asym_line_input.x_cb);
+            z_series_abc = r_matrix + x_matrix * 1.0i;
+        } 
+        else {
+            ComplexTensor4 r_matrix = ComplexTensor4(asym_line_input.r_aa, asym_line_input.r_bb, asym_line_input.r_cc, asym_line_input.r_nn, asym_line_input.r_ba, asym_line_input.r_ca, asym_line_input.r_na, asym_line_input.r_cb, asym_line_input.r_nb, asym_line_input.r_nc);
+            ComplexTensor4 x_matrix = ComplexTensor4(asym_line_input.x_aa, asym_line_input.x_bb, asym_line_input.x_cc, asym_line_input.x_nn, asym_line_input.x_ba, asym_line_input.x_ca, asym_line_input.x_na, asym_line_input.x_cb, asym_line_input.x_nb, asym_line_input.x_nc);
+
+            ComplexTensor4 y = r_matrix + 1.0i * x_matrix;
+            z_series_abc = kron_reduction(y);
+        }
+        ComplexTensor<asymmetric_t> a_matrix = get_sym_matrix();
+        ComplexTensor<asymmetric_t> a_matrix_inv = get_sym_matrix_inv();
+        ComplexTensor<asymmetric_t> z_series = (a_matrix_inv.matrix() * z_series_abc.matrix() * a_matrix.matrix()).array();
+        return z_series;
+    }
+
+    ComplexTensor<asymmetric_t> compute_c_matrix_from_input(const power_grid_model::AsymLineInput& asym_line_input) {
+        ComplexTensor<asymmetric_t> c_matrix;
+        if (!is_nan(asym_line_input.c0) && !is_nan(asym_line_input.c1)) {
+            c_matrix = ComplexTensor<asymmetric_t>(asym_line_input.c0 + asym_line_input.c1, -asym_line_input.c1);
+        } 
+        else if (is_nan(asym_line_input.c_nn)) {
+            c_matrix = ComplexTensor<asymmetric_t>(asym_line_input.c_aa, asym_line_input.c_ba, asym_line_input.c_bb, asym_line_input.c_ca, asym_line_input.c_cb, asym_line_input.c_cc);
```

Comment:
This is not the same order as you define the constructor in the `three_phase_tensor.hpp`. Have a look again.